fix: finish migrating from setup.cfg to pyproject.toml in order to install seqrepo cli

reece opened the pull request:
See #138 for explanation.

@biocommons/maintainers : To test, please pull this branch and `make devready` then try `seqrepo`. If that works, I think we're good to go.

korikuzma reviewed (changes requested):
I'm having trouble running `make devready` on a Mac:

```
make venv && source venv/bin/activate && make develop
/bin/bash: line 0: globstar: invalid shell option name
make: *** [devready] Error 2
```

Changing: `SHELL:=/bin/bash -e -o pipefail -O globstar` to `SHELL:=/bin/bash -e -o pipefail` gets `make devready` to work.

I then run `source venv/bin/activate`. After running `seqrepo`, I get:
```
Traceback (most recent call last):
  File "/Users/kxk102/Documents/biocommons/biocommons.seqrepo/venv/bin/seqrepo", line 5, in <module>
    from biocommons.seqrepo.cli import main
  File "/Users/kxk102/Documents/biocommons/biocommons.seqrepo/src/biocommons/seqrepo/cli.py", line 34, in <module>
    import tqdm
ModuleNotFoundError: No module named 'tqdm'
```

We'll want to update the `develop` in `Makefile` to `pip install -e .[dev,docs,loading]`.

After these changes, I can successfully run `seqrepo`.

reece commented:
I removed the loading optional deps (it was new) and put its deps in dev.

I also added instructions for Mac OSX, which includes installing bash. (Although the makefile appeared to work for you, the clean targets that use globstar would fail. globstar needs to stay.

@korikuzma Would you please try again?

korikuzma reviewed (approved):
Can confirm those changes work! Thanks @reece
